test/s3path.jl

```diff
@@ -354,4 +355,39 @@ end
     rm(json_path)
 end
 
+@testset "S3Path versioning" begin
+    s3_enable_versioning(aws, bucket_name)
+    key_version_file = "test_versions"
+    s3_put(aws, bucket_name, key_version_file, "data.v1")
+    s3_put(aws, bucket_name, key_version_file, "data.v2")
+
+    # `s3_list_versions` returns versions in the order newest to oldest
+    versions = [d["VersionId"] for d in reverse!(s3_list_versions(aws, bucket_name, key_version_file))]
+    @test length(versions) == 2
+    @test read(S3Path(bucket_name, key_version_file; config=aws, version=first(versions)), String) == "data.v1"
+    @test read(S3Path(bucket_name, key_version_file; config=aws, version=last(versions)), String) == "data.v2"
+    @test isequal(read(S3Path(bucket_name, key_version_file; config=aws, version=last(versions)), String),
+                  read(S3Path(bucket_name, key_version_file; config=aws), String))
+    @test isequal(read(S3Path(bucket_name, key_version_file; config=aws, version=last(versions)), String),
+                  read(S3Path(bucket_name, key_version_file; config=aws, version=""), String))
```

Comment:
Using `nothing` should be the preferred way to get the latest version
```suggestion
                  read(S3Path(bucket_name, key_version_file; config=aws, version=nothing), String))
```